server.lua

```diff
@@ -36,26 +36,80 @@ local function UniqueAccounts(player)
 			}
 			exports.pefcl:createUniqueAccount(playerSrc, data)
 		end
-
-		local role = false
-		if PlayerJob.grade.level >= Config.BusinessAccounts[PlayerJob.name].AdminRole then
-			role = 'admin'
-		elseif PlayerJob.grade.level >= Config.BusinessAccounts[PlayerJob.name].ContributorRole then
-			role = 'contributor'
-		end
-
-		if role then
-			local data = {
-				role = role,
-				accountIdentifier = PlayerJob.name,
+	end
+	local accounts = exports.pefcl:getAccounts(playerSrc).data
+	for k,v in pairs(accounts) do
+		if Config.BusinessAccounts[v.ownerIdentifier] and v.ownerIdentifier == PlayerJob.name then
+			local role = false
+			if PlayerJob.grade.level >= Config.BusinessAccounts[v.ownerIdentifier].AdminRole then
+				role = 'admin'
+			elseif PlayerJob.grade.level >= Config.BusinessAccounts[v.ownerIdentifier].ContributorRole then
+				role = 'contributor'
+			end
+			if not role then
+				local data1 = {
+					userIdentifier = citizenid,
+					accountIdentifier = v.ownerIdentifier,
+				}
+				exports.pefcl:removeUserFromUniqueAccount(playerSrc, data1)
+			elseif v.role ~= role then
+				local data1 = {
+					userIdentifier = citizenid,
+					accountIdentifier = v.ownerIdentifier,
+				}
+				exports.pefcl:removeUserFromUniqueAccount(playerSrc, data1)
+				if Config.BusinessAccounts[PlayerJob.name] then
+					local role = false
+					if PlayerJob.grade.level >= Config.BusinessAccounts[PlayerJob.name].AdminRole then
+						role = 'admin'
+					elseif PlayerJob.grade.level >= Config.BusinessAccounts[PlayerJob.name].ContributorRole then
+						role = 'contributor'
+					end
+					if role then
+						local data = {
+							role = role,
+							accountIdentifier = PlayerJob.name,
+							userIdentifier = citizenid,
+							source = playerSrc
+						}
+						exports.pefcl:addUserToUniqueAccount(playerSrc, data)
+					end
+				end
+			end 
+		elseif Config.BusinessAccounts[v.ownerIdentifier] and v.ownerIdentifier ~= PlayerJob.name then
+			local data1 = {
+				userIdentifier = citizenid,
+				accountIdentifier = v.ownerIdentifier,
+			}
+			exports.pefcl:removeUserFromUniqueAccount(playerSrc, data1)
+		elseif Config.BusinessAccounts[PlayerJob.name] and v.ownerIdentifier ~= PlayerJob.name then
```

Comment:
```suggestion
		elseif Config.BusinessAccounts[PlayerJob.name] and v.ownerIdentifier ~= PlayerJob.name and v.type == 'shared' then
```

Threw this on my server and it was throwing a log trying to remove the personal account